apicapi/apic_client.py

```diff
@@ -490,6 +519,28 @@ def _save_cookie(self, request, response):
             attributes = imdata[0]['error']['attributes']
         return attributes
 
+    def _calculate_signature(self, request, url, kwargs):
+        """ Sign the request using private key and put the result in cookies"""
+        if self._is_cert_auth():
+            if url.endswith('?'):
+                url = url[:-1]
+            url = url.replace('//', '/')
+            payLoad = request + '/api' + url + kwargs.get('data', '')
+            signedDigest = sign(self.private_key, payLoad, 'sha256')
```

Comment:
is the hash function something fixed? Any chance we may need to change it with new APIC versions/configurations? If so we may consider making it configurable (same goes for L533)